app/controllers/health_profiles_controller.rb

```diff
@@ -0,0 +1,48 @@
+class HealthProfilesController < ApplicationController
+	before_action: require_login
+	
+
+	def index
+	end
+
+	def new
+		# @health_profile = current_user.patients.find(params[:id])
+		@health_profile = HealthProfile.new
+	end
+
+	def create
+		# @health_profile = current_user.health_profiles.new(health_profile_params)
+		@health_profile = HealthProfile.new
+		if @health_profile.save
+			redirect_to health_profile_path, notice: "Your profile has been created."
+		else
+			flash[:notice] = "Error creating new profile."
+			render 'new'
+		end
+
+	end
+
+	def show
+		# @health_profile = current_user.health_profiles.find(params[:id])
+		@health_profile = HealthProfile.find(params[:id])
+		@allergies = health_profile.allergies.find(params[:patient_id])
+		@immunizations = health_profile.immunizations.find(params[:patient_id])
+		@prescriptions = health_profile.prescriptions.find(params[])
```

Comment:
`.find` returns first result found- if we are looking for multiple, I think we need to use `.where`